UsefullExtensionLib/usefullExtensionLibrary/src/main/java/com/nbprog/usefullextensionlibrary/ValidationExtension.kt

```diff
@@ -0,0 +1,13 @@
+package com.nbprog.usefullextensionlibrary
+
+import java.util.regex.Pattern
+
+/**
+ * Check is Email valid
+ * Def expression value : "^\w+([-+.']\w+)*@\w+([-.]\w+)*\.\w+([-.]\w+)*$"
+ * */
+fun String.isEmailValid(expression: String = "^\\w+([-+.']\\w+)*@\\w+([-.]\\w+)*\\.\\w+([-.]\\w+)*\$"): Boolean {
+    val pattern = Pattern.compile(expression, Pattern.CASE_INSENSITIVE)
+    val matcher = pattern.matcher(this)
+    return matcher.matches()
```

Comment:
można to znacznie uprościć, po pierwsze regexy przyjmuj w """ regex """ -> pozwoli to na dosłowne przepisywanie regexów np ze stron itp. nie trzeba dbać o pisanie "\" przed każdym wyrażeniem ktore chcemy brać dosłownie. Dodatkowo czemu parametr jest typu String a nie Regex? wnętrze funkcji można generalnie skrócić do postaci:

expression.toRegex().matches(this)

Reply:
Poprawione